### PR TITLE
Fixed #32110 -- Doc'd and tested enumerations for ChoiceField.choices.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: tests/forms_tests/field_tests/test_choicefield.py [hunk 1]
```diff
@@ -87,3 +88,14 @@ def test_choicefield_disabled(self):
...
             '<option value="P">Paul</option></select>'
         )
+
+    def test_choicefield_enumeration(self):
+        class FirstNames(models.TextChoices):
+            JOHN = 'J', 'John'
+            PAUL = 'P', 'Paul'
+
+        f = ChoiceField(choices=FirstNames.choices)
+        self.assertEqual('J', f.clean('J'))
```
Comment (pope1ni) [inline tests/forms_tests/field_tests/test_choicefield.py:98]: So we should probably use the enum here. Something like:
```suggestion
        self.assertEqual(FirstNames.JOHN.value, f.clean(FirstNames.JOHN))
```

**Review by pope1ni — APPROVED** [review]
Thanks! So there is a already a general link through for more detail, but being explicit is more helpful.

I guess this could be backported through 3.0 as it is a documentation oversight and we're not adding anything new?

File: tests/forms_tests/field_tests/test_choicefield.py (hunk 1 above)
Comment (felixxm) [reply tests/forms_tests/field_tests/test_choicefield.py:98]: I think the current form is fine. I would leave it.

**Review by felixxm — COMMENTED** [review]
@claudep Thanks :+1:

**Pull request merged**